### DISCUSSION TITLE
本博客模板常见问题 Q & A — 码志

### DISCUSSION THREAD

**Discussion by mzlogin** [mzlogin/blog-comments#204] [Announcements]
https://mazhuang.org/2020/05/03/blog-template-qna/ 

 使用这个博客模板的朋友们时不时会提出一些问题，我将它们以及对应的解决方案逐渐整理归纳，汇总到这一篇帖子里。

**Comment by sunsetroads** [comment]
代码高亮的风格有办法改吗？

**Reply by myxinyue** [reply]
为什么我删除了你的文件，打开网页还是能显示

**Comment by mzlogin** [comment]
> @sunsetroads
> 代码高亮的风格有办法改吗？

今天刚刚改造了一下，现在可以通过 _config.yml 文件里的配置项 `highlight_theme` 来指定代码高亮风格，支持的风格名称列表参考我维护的另一个项目：

- <https://github.com/mzlogin/rouge-themes>

在项目主页可以看到每种风格的预览效果。

以上内容已经补充进 Q&A 正文里。

**Comment by ghost** [comment]
你好,想要问一下,这个本地图片要怎么插入到markdown上,我把图片插入到写的文章中无法显示,文件我都是传到根目录的images里面的,然后使用相对路径,图片就是无法显示,不知道是哪里出错了,现在的解决办法是把图片上传到图床很麻烦

**Comment by mzlogin** [comment]
> @dutee
> 你好,想要问一下,这个本地图片要怎么插入到markdown上,我把图片插入到写的文章中无法显示,文件我都是传到根目录的images里面的,然后使用相对路径,图片就是无法显示,不知道是哪里出错了,现在的解决办法是把图片上传到图床很麻烦

放在根目录的 images 里可以参考我的写法：

```
![](/images/xxx/yyy.png)
```

**Comment by bsxw** [comment]
gitalk怎么弄，没留言还是差了点

**Comment by mzlogin** [comment]
> @bsxw
> gitalk怎么弄，没留言还是差了点

我看你已经改好了。_config.yml 第 113 行左右的链接里有 gitalk 的安装说明。

**Comment by mzlogin** [comment]
> @Galaxy-Studio-Code
> 请问改了favicon长时间不生效怎么解决？

这个一般是浏览器缓存问题，换个浏览器看看效果，或者在网上搜索一下清除 favicon 缓存的方法。

**Comment by qisi007** [comment]
评论咋整的

**Comment by mzlogin** [comment]
> @qisi007
> 评论咋整的

用的 gitalk，配置方法在 _config.yml 文件里有参考链接：https://github.com/gitalk/gitalk#install

**Comment by cheng3100** [comment]
hi， 请问如何配置首页的blog的摘要范围？我的一些博客的摘要有的被显示出来非常长，有的就还好,就是指的在首页的blog的摘要显示

**Comment by mzlogin** [comment]
> hi， 请问如何配置首页的blog的摘要范围？我的一些博客的摘要有的被显示出来非常长，有的就还好,就是指的在首页的blog的摘要显示

如果是按我目前的配置，是显示第一段。所以如果你摘要很长，分个段…… 😆

**Comment by cheng3100** [comment]
> > hi， 请问如何配置首页的blog的摘要范围？我的一些博客的摘要有的被显示出来非常长，有的就还好,就是指的在首页的blog的摘要显示
> 
> 如果是按我目前的配置，是显示第一段。所以如果你摘要很长，分个段…… 😆

能帮忙看下吗？我对比了我的博客里的确是分段了，但不知道为什么有几篇摘要特别长

**Comment by cheng3100** [comment]
> @cheng3100
> > > hi， 请问如何配置首页的blog的摘要范围？我的一些博客的摘要有的被显示出来非常长，有的就还好,就是指的在首页的blog的摘要显示
> > 
> > 如果是按我目前的配置，是显示第一段。所以如果你摘要很长，分个段…… 😆
> 
> 能帮忙看下吗？我对比了我的博客里的确是分段了，但不知道为什么有几篇摘要特别长

已解决，jekyll的摘要分隔符为excerpt_separator，我看到你的设置为`\n\n`， 这个会在跨平台编辑时出现问题，因为windows和mac平台下的换行符没有连续的‘\n‘ 出现的，如果在windows的一些编辑器里编辑时就会找不到这个分隔符。这个地方建议修改下默认配置为“======”等比较好

**Comment by mzlogin** [comment]
> > @cheng3100
> > > > hi， 请问如何配置首页的blog的摘要范围？我的一些博客的摘要有的被显示出来非常长，有的就还好,就是指的在首页的blog的摘要显示
> > > 
> > > 
> > > 如果是按我目前的配置，是显示第一段。所以如果你摘要很长，分个段…… 😆
> > 
> > 
> > 能帮忙看下吗？我对比了我的博客里的确是分段了，但不知道为什么有几篇摘要特别长
> 
> 已解决，jekyll的摘要分隔符为excerpt_separator，我看到你的设置为`\n\n`， 这个会在跨平台编辑时出现问题，因为windows和mac平台下的换行符没有连续的‘\n‘ 出现的，如果在windows的一些编辑器里编辑时就会找不到这个分隔符。这个地方建议修改下默认配置为“======”等比较好

感谢建议，原来是这个原因。这个问题之前有人提过：<https://github.com/mzlogin/mzlogin.github.io/issues/47> 讨论过。

可以配置 git 提交时自动转换换行符，也可以改为使用 description 等，用 `=========` 会影响正文显示，你可以用注释将它包起来，如 `<!-- ======= -->`。我考虑下如何优雅点解决。

**Comment by wvalianty** [comment]
大佬，现在评论不好用了，怎么办啊，https://cors-anywhere.herokuapp.com/corsdemo

**Comment by mzlogin** [comment]
> 大佬，现在评论不好用了，怎么办啊，https://cors-anywhere.herokuapp.com/corsdemo

gitalk 的项目 Issues 有很多关于这个问题的讨论 <https://github.com/gitalk/gitalk/issues?q=is%3Aissue+is%3Aclosed> ，维护者已经更新了版本来解决此问题，在讨论里也可以看到其它解决方案。

如果你是使用本模板的最新代码，那其实不用做什么，会自动引用最新版本。如果一再刷新后还是不行的话，那需要刷新一下你本地的缓存，方法：

访问以下 2 个链接：

- <https://purge.jsdelivr.net/npm/gitalk@1/dist/gitalk.min.js>
- <https://cdn.jsdelivr.net/npm/gitalk@1/dist/gitalk.min.js?v=1.7.2>

**Comment by mzlogin** [comment]
> @jlukey
> 您好博主，请问jekyll里面默认的日期会转成英文格式，你是咋处理的呀？
> 比如markdown头信息中这样定义：2021-04-15
> 但是在页面加载出来就显出为：Apr 15, 2021
> 

可以参考 <https://liquid.bootcss.com/filters/date/> 和 <https://github.com/mzlogin/mzlogin.github.io/blob/b01aa72fe13e868ae6604cef7359218540133a46/_layouts/post.html#L18>

**Comment by kidoneself** [comment]
老哥，我全局替换的域名，但是搜索出来的东西还是跳转的你博客，这个搜索在哪儿改呢

**Comment by mzlogin** [comment]
> 老哥，我全局替换的域名，但是搜索出来的东西还是跳转的你博客，这个搜索在哪儿改呢

你先把域名配置对了试下，我看你现在配置的是一个 IP 地址，这样不行的吧，要么绑定域名，要么用 GitHub 自带的域名。

**Comment by kidoneself** [comment]
好的，我已经解决了，修改了代码，我用的webhook部署的国内的，域名正在备案中，先用ip配置起来的。
谢谢老哥的模板！

**Comment by kidoneself** [comment]
顺带问一下，你这个邮箱发送邮件的怎么配置的，是不是有人评论你会收到邮件，你回复了我这也收到邮件了

**Comment by mzlogin** [comment]
> @kidoneself
> 顺带问一下，你这个邮箱发送邮件的怎么配置的，是不是有人评论你会收到邮件，你回复了我这也收到邮件了

是的。这个是 gitalk 利用 GitHub Issues 做评论框，GitHub Issues 带的功能，把 gitalk 配置好就行了。

**Comment by liaoj12** [comment]
大大好，请问一下哪里可以修改整个blog的theme？我不知道修改哪里了，现在这个blog的theme都崩了。。。

**Comment by mzlogin** [comment]
> 大大好，请问一下哪里可以修改整个blog的theme？我不知道修改哪里了，现在这个blog的theme都崩了。。。

看不到你的仓库源码，你要是不知道怎么处理，重新按 README 里的 Fork 指南来一遍吧……

**Comment by liaoj12** [comment]
你好大大，我重新弄了一下后发现是在config里的main configs里，只要把你的url从"https://mazhuang.org"改成"https://liaoj12.github.io/"就会没有theme的css了。我需要先安装点什么吗？感谢

**Comment by mzlogin** [comment]
> 你好大大，我重新弄了一下后发现是在config里的main configs里，只要把你的url从"[https://mazhuang.org"改成"https://liaoj12.github.io/"就会没有theme的css了。我需要先安装点什么吗？感谢](https://mazhuang.org%22%E6%94%B9%E6%88%90%22https://liaoj12.github.io/%22%E5%B0%B1%E4%BC%9A%E6%B2%A1%E6%9C%89theme%E7%9A%84css%E4%BA%86%E3%80%82%E6%88%91%E9%9C%80%E8%A6%81%E5%85%88%E5%AE%89%E8%A3%85%E7%82%B9%E4%BB%80%E4%B9%88%E5%90%97%EF%BC%9F%E6%84%9F%E8%B0%A2)

url 和 repository 一起改掉，应该不会有什么问题。

**Comment by liaoj12** [comment]
好了 谢谢大大

**Comment by JBoRu** [comment]
你好，我想删除 维基 那个页面 并重新新建一个论文 但是本地编译时总会说 没有论文这个页面 我应该查看哪些文件做修改呢？（我现在比较笨的方法是直接修改了pages/wiki.md内容）感谢回复！

**Comment by mzlogin** [comment]
> 你好，我想删除 维基 那个页面 并重新新建一个论文 但是本地编译时总会说 没有论文这个页面 我应该查看哪些文件做修改呢？（我现在比较笨的方法是直接修改了pages/wiki.md内容）感谢回复！

修改 _config.yml 79-81 行。

**Comment by leleji** [comment]
666

**Comment by LfqGithub** [comment]
请问，如果不想要blog这一栏，打开首页就是wiki, 应该如何设置？ 谢谢！

**Comment by mzlogin** [comment]
> @LfqGithub
> 请问，如果不想要blog这一栏，打开首页就是wiki, 应该如何设置？ 谢谢！

1. 删掉 index.html；
2. 将 pages/wiki.md 复制到根目录下，命名 index.md；
3. 将 index.md 里的 permalink 改为 `/`。

**Comment by LfqGithub** [comment]
> @mzlogin
> > @LfqGithub
> > 请问，如果不想要blog这一栏，打开首页就是wiki, 应该如何设置？ 谢谢！
> 
> 1. 删掉 index.html；
> 2. 将 pages/wiki.md 复制到根目录下，命名 index.md；
> 3. 将 index.md 里的 permalink 改为 `/`。


收到，感谢！

**Comment by anglelikepupu** [comment]
请问这个能运行到别的服务器上吗，还是只能用在GitHub上？

**Comment by mzlogin** [comment]
> @anglelikepupu
> 请问这个能运行到别的服务器上吗，还是只能用在GitHub上？

使用 Jekyll 编译完之后就是纯静态资源，在各种 Web 服务器上应该都能部署。

**Comment by QinZXX** [comment]
请问分类的设置在哪里操作呢？修改分类，删除和添加分类怎么做呢

**Comment by mzlogin** [comment]
> @QinZXX
> 请问分类的设置在哪里操作呢？修改分类，删除和添加分类怎么做呢

每篇文章头部的 categories 指定好即可，会自动处理，不用单独维护分类。

**Comment by QinZXX** [comment]
> @mzlogin
> > @QinZXX
> > 请问分类的设置在哪里操作呢？修改分类，删除和添加分类怎么做呢
> 
> 每篇文章头部的 categories 指定好即可，会自动处理，不用单独维护分类。

明白了，谢谢！

**Comment by lxb7267247** [comment]
测试

**Comment by kidneybean-beef** [comment]
请问文章标题背景里的二维码图片可以去掉或者改成自己的吗？

**Comment by QinZXX** [comment]
您发给我的邮件已经收到。This is an automatic reply, confirming that your e-mail was received.Thank you

**Comment by mzlogin** [comment]
> @kidneybean-beef
> 请问文章标题背景里的二维码图片可以去掉或者改成自己的吗？

可以的。_config.yml 里 components > qrcode 这一段 就是控制这个的。

去掉：将 components.qrcode.enabled 改为 false

改成自己的：替换 assets/images/qrcode.jpg 文件

**Comment by tangyun-92** [comment]
老哥，怎么我的评论下面显示：Error: Not Found.，然后我评论，控制台直接报了404！
ttps://api.github.com/repos/tangyun-92/tangyun-blog/issues?labels=gitment,%2F2020%2F11%2F11%2FVue%25E9%25A1%25B9%25E7%259B%25AE%25E9%25A6%2596%25E5%25B1%258&t=1640158668129 404

**Comment by tangyun-92** [comment]
感谢老哥通过微信帮我解决了问题！！！

**Comment by tangninghaha** [comment]
您好博主，请问如何改图标呢？

**Comment by mzlogin** [comment]
> @tangninghaha
> 您好博主，请问如何改图标呢？
> 

浏览器标签栏的图标：替换 favicon.ico 文件。

网站左上角图标：修改 _includes/header.html#L86 的 octicon octicon-mark-github 那块。

**Comment by tyjwan** [comment]
> > @Galaxy-Studio-Code
> > 请问改了favicon长时间不生效怎么解决？
> 
> 这个一般是浏览器缓存问题，换个浏览器看看效果，或者在网上搜索一下清除 favicon 缓存的方法。

我遇到过，是因为，博客的所有内容都会放到cdn里加速，然后图片文件和.ico文件，如果文件名一样，cdn不会替换，我已经验证过了。。我是通过改文件名来解决这个问题的

**Comment by QinZXX** [comment]
您发给我的邮件已经收到。This is an automatic reply, confirming that your e-mail was received.Thank you

**Comment by antsmallant** [comment]
奇怪，我的 giscus 配置了，但没法评论，在这里试一下：）

**Reply by zkitefly** [reply]
评论需要登录github账户授权giscus权限才能在页面上发评论哦

**Comment by myxinyue** [comment]
为什么我删除了文件，打开后还是您的那些文章

**Reply by mzlogin** [reply]
我看你的页面现在已经正常，文章都已经删除了啊。
一般如果删除了内容没有更新，有可能是编译失败，可以在GitHub页面上看最近一次的提交信息那有没有红色小叉，点开能看到详细信息。

**Comment by levenwindy** [comment]
你好，如果想将背景颜色调成黑色，应该在哪里修改？index.html吗

**Reply by mzlogin** [reply]
可以在 assets/css/globals/common.css 里指定样式。

**Reply by levenwindy** [reply]
> 可以在 assets/css/globals/common.css 里指定样式。

感谢

**Comment by Dogkun** [comment]
我的建完之后就像依托答辩，毫无样式

**Reply by Dogkun** [reply]
可以帮忙看看嘛
http://dogkun.ml/blog/

**Reply by mzlogin** [reply]
我看你已经弄好了啊。

**Reply by WalnutTTTT-Qiao** [reply]
我的样式刚上去的时候有，现在没了

**Reply by WalnutTTTT-Qiao** [reply]
找到原因了我把jsDelivr CDN打开 之后加载的路径出现了重复，我想问下 老哥路由配置是在哪的呀

**Reply by mzlogin** [reply]
@Fengqiaoshu 路由配置，具体指什么？

**Reply by WalnutTTTT-Qiao** [reply]
我打开cdn 然后加载css样式的路径就多一个段
https://cdn.jsdelivr.net/gh/https://github.com/Fengqiaoshu/myglod.git@master/assets/css/global  
我想问问是不是有什么能统一改路径的地方 @mzlogin

**Reply by mzlogin** [reply]
@Fengqiaoshu _config.yml 文件里的 repository 配置项去掉 `https://github.com/`。

路径的拼接在 _layouts/default.html 文件里可以看到。

**Reply by WalnutTTTT-Qiao** [reply]
@mzlogin   好的 感谢老哥

**Comment by xdd1997** [comment]
您好，我的网站也是想使用giscus，目前存在的问题是，在网页评论后，评论内容能显示在discussions，但是评论内容无法直接显示在评论的网页上，请问您有什么解决经验吗，谢谢

**Reply by mzlogin** [reply]
我去你的网站试了下，category 和 category-id 配置的不匹配。

你的 category 配置的 Announcements，但 category-id 配置的是对应 General 的，看现象，这导致创建 Discussion 的时候创建在 General 分类下，但查找时去 Announcements 分类下查找找不到。

你把它们改成一致了再试试。

**Reply by xdd1997** [reply]
非常非常感谢，祝您新年快乐~

**Comment by wangyuzhen666** [comment]
请问，打开我的首页后页面右上角的“首页”及其他类似链接都是链接到https://mazhuang.org/，请问怎么调成我的？

**Reply by mzlogin** [reply]
把 _config.yml 的 url 改成你的就行了，我看你已经改了。但是你的配置文件里有错误导致编译失败了。

错误原因见：https://github.com/wangyuzhen666/wangyuzhen666.github.io/actions/runs/4070402685/jobs/7011211170

里面指出了错误是在 _config.yml 的 line 74 column 5，label 拼写错了，冒号改成英文符号应该就好了。

**Reply by wangyuzhen666** [reply]
可是改过来后仍然不行怎么办？

**Reply by mzlogin** [reply]
我看现在好了啊。首页仍然有问题是因为 _config.yml 第 74 行 label 拼写错了。

**Comment by wangyuzhen666** [comment]
请问如何设置一个Markdown为置顶？谢谢！

**Reply by mzlogin** [reply]
在文件头部的元信息里指定 `topmost: true`

<img width="750" alt="image" src="https://user-images.githubusercontent.com/1646590/216495045-cfabd732-a029-4f66-a3b8-94d5cedcac7e.png">

**Comment by wangyuzhen666** [comment]
请问如何在网站首页上添加公告栏浮窗？谢谢

**Reply by mzlogin** [reply]
目前无此功能。个人主页感觉必要性不大，如果确实觉得有必要可以自行开发。

**Comment by wangyuzhen666** [comment]
GitHubpage 有两篇文章上传到了_post中，但站点却无法显示。

**Reply by mzlogin** [reply]
<img width="664" alt="image" src="https://user-images.githubusercontent.com/1646590/219233861-15cadffc-5754-456a-9bd4-4e2cfde2e1eb.png">

检查下你的提交记录，从这次提交后，后面一直都编译失败了，点这个红色的叉可以看到是哪个文件报错。可以尝试恢复引起报错的那次提交的修改看看编译能否通过。

**Comment by wangyuzhen666** [comment]
GitHubpage 有两篇文章上传到了_post中，编译也成功了，但站点却无法显示。

**Reply by mzlogin** [reply]
检查下那两个文件名跟其它的有什么不同，看着应该是因为没加 .md 后缀。
<img width="685" alt="image" src="https://user-images.githubusercontent.com/1646590/220551711-0bcde009-180a-4321-9a2b-b8a2608f4ee4.png">

**Comment by wangyuzhen666** [comment]
githubpage仓库中的文件，在浏览器中输入路径后不能显示。例如输入：wangyuzhen666.github.io/pages/flash.html
可却不能正常显示

**Reply by wangyuzhen666** [reply]
抱歉，不需要了，谢谢

**Comment by westqzy** [comment]
博客好像不显示表格欸，想问下有没有解决的办法。
| **1** | **2** | **3** |
|:-----:|:-----:|:-----:|
|   1   |   1   |   1   |
|   1   |   1   |   1   |

目前是通过在_layout/post.html中加  <style>table{border-left:1px solid #000000;border-top:1px solid #000000;width: 100%;word-wrap:break-word; word-break:break-all;}table th{text-align:center;}table th,td{border-right:1px solid #000000;border-bottom:1px solid #000000;}
</style>

**Reply by mzlogin** [reply]
表格显示没问题啊，示例：

https://mazhuang.org/wiki/markdown/#%E8%A1%A8%E6%A0%BC

**Reply by westqzy** [reply]
我找到原因了我自己排版排错了。。太太太感谢了

**Reply by mzlogin** [reply]
不客气～

**Comment by carpedx** [comment]
为什么友链links.yml name不能是中文，加入中文就Build报错

/usr/local/bundle/gems/safe_yaml-1.0.5/lib/safe_yaml/load.rb:143:in `parse': (/github/workspace/_data/links.yml): invalid leading UTF-8 octet at line 1 column 1 (Psych::SyntaxError)

**Reply by mzlogin** [reply]
<img width="412" alt="image" src="https://github.com/mzlogin/blog-comments/assets/1646590/4c37d983-ade5-4f62-b73f-ce159958306c">

links.yml name 可以是中文，你遇到这个问题应该是你编辑文件时的编码问题，我看你之前编译出错的时候 name 显示乱码。

https://github.com/carpedx/carpedx.github.io/blob/6faace7cf7/_data/links.yml

**Comment by hatefish114514** [comment]
存在文件夹里的图片在博客上不显示，但本地博客预览可以看到

**Reply by hatefish114514** [reply]
solved

**Comment by hatefish114514** [comment]
wiki文章不显示

**Reply by hatefish114514** [reply]
solved

**Comment by IDONT5WAP** [comment]
怎么将片段,或者wiki 修改成其他的栏目,
我单纯改了config里的href,label 感觉好像不起效果.还应该改哪些文件?

**Reply by mzlogin** [reply]
改这就行了。检查下对应的 href 路径是否存在，推送之后编译是否通过。

**Comment by xueche** [comment]
大佬，我替换了_config.yml中的url和reposity之后，另外把“comments_provider”字段置为空。然后网站的布局格式就没有，可以帮忙看看是哪里的问题吗？这是我的代码仓库：https://github.com/xueche/xueche.github.io/tree/master

**Reply by mzlogin** [reply]
<img width="1533" alt="image" src="https://github.com/mzlogin/blog-comments/assets/1646590/a7711aa9-ee3f-40bf-a817-72c5a274ff95">


可以 F12 看下开发者工具控制台，我看你的仓库里 assets/vendor 目录被删了，有一些静态资源找不到，导致页面加载中断。

**Reply by xueche** [reply]
问题已解决，是.gitignore中将vendor目录给忽略了，感谢大佬~

**Comment by wangyuzhen666** [comment]
文件编译成功了，但网站却没有出现该文章，怎么办？文章在仓库中链接：[问题文章的仓库链接https://...](https://github.com/wangyuzhen666/wangyuzhen666.github.io/blob/main/_posts/2023-8-1-%E7%A7%91%E6%8A%80%E5%BC%BA%E5%86%9B%E5%92%8C%E4%BF%A1%E6%81%AF%E6%88%98%E5%AE%89%E5%85%A8%E4%B9%8B%E4%BB%A3%E7%A0%81%E8%AE%BA.md)
                                还有我的仓库：https://github.com/wangyuzhen666/wangyuzhen666.github.io

**Reply by wangyuzhen666** [reply]
网站：wangyuzhen666.github.io

**Reply by mzlogin** [reply]
<img width="1362" alt="image" src="https://github.com/mzlogin/blog-comments/assets/1646590/a8a491fa-1608-435b-aafe-dacb1c99b72e">

看内容没什么问题。看编译日志，应该是因为编译服务器上的时区与我们的不一样，它当时的时间还在2023-07-31（右上角有个设置按钮，可以显示时间戳），认为你的8月1日的文章是属于未来时间的，所以跳过了。

你现在重新触发下编译应该可以出来了。

**Reply by wangyuzhen666** [reply]
谢谢

**Comment by wangyuzhen666** [comment]
我在github上删除了一篇自己的 post文章  https://kgithub.com/wangyuzhen666/wangyuzhen666.github.io/commit/5ba07e642304d7cb3bee05f777d3fd19bd4b000e/  ，但是当删除完成后，网站的首页上却仍然出现了该文章的索引，但是当点开后，却发现显示404页面未找到。我明明已经删除了可是仍然有他的索引。请问该怎么办？谢谢！

**Reply by wangyuzhen666** [reply]
![](wangyuzhen666.github.io/test/屏幕截图 2023-08-01 171331.png)

**Reply by wangyuzhen666** [reply]
solved ,thankyou

**Reply by wangyuzhen666** [reply]
收回我刚刚说的话，没解决，呜呜呜

**Reply by mzlogin** [reply]
你首页看不到了啊，浏览器本地缓存吧？

另外，你评论里贴的地址非 GitHub 官方，是 kgithub.com，注意用户安全。

**Comment by jerry1233** [comment]
请问文章结尾处的“文档信息-本文作者”应该如何修改？在哪个配置文件？

**Reply by mzlogin** [reply]
在文件 _includes/copyright.html，修改对应的内容或者配置项即可。

**Comment by lanruixiang** [comment]
这个博客模版能否使用 $\LaTeX$

**Reply by mzlogin** [reply]
可以的，集成了 MathJax，不过默认不是全局开启的，需要在要开启的页面文件的头部指定 `mathjax: true`，例如：

https://github.com/mzlogin/mzlogin.github.io/blob/master/_wiki/markdown.md?plain=1#L11

显示效果：

https://mazhuang.org/wiki/markdown/#mathjax

**Reply by lanruixiang** [reply]
谢谢大佬

**Reply by lanruixiang** [reply]
明白了

**Comment by zhipengliux** [comment]
请问这个模板如何插入视频和音频呢

**Reply by zhipengliux** [reply]
<video src="https://www.youtube.com/watch?v=IGJMb95--m8&list=PL_bgmYHGITt-8Noh1fZpxTHJ-GrBb626R&index=27&t=233s"></video>这样弄，但并不显示

**Reply by zhipengliux** [reply]
<video src="https://www.youtube.com/watch?v=IGJMb95--m8&list=PL_bgmYHGITt-8Noh1fZpxTHJ-GrBb626R&index=27&t=233s"></video>

**Reply by zhipengliux** [reply]
``` html
<video src="https://www.youtube.com/watch?v=IGJMb95--m8&list=PL_bgmYHGITt-8Noh1fZpxTHJ-GrBb626R&index=27&t=233s"></video>
```

**Reply by zhipengliux** [reply]
``` html
<iframe src="https://www.youtube.com/watch?v=IGJMb95--m8&list=PL_bgmYHGITt-8Noh1fZpxTHJ-GrBb626R&index=27&t=233s"></iframe>
```
改成这样但是显示拒绝连接

**Reply by zhipengliux** [reply]
B站可以，youtube不可以

**Reply by mzlogin** [reply]
这个就是直接用 HTML 写，跟用什么模板没关系了。youtube 不行估计是得科学上网吧。

**Comment by zhipengliux** [comment]
主页顶部那个蓝色的方块在哪里换颜色呀

**Reply by mzlogin** [reply]
assets/css/pages/index.css 第3行。

可以 F12 看浏览器控制台来定位。

**Comment by carpedx** [comment]


**Reply by lanruixiang** [reply]
我感觉是先按照 cate1 的字典序排列，再按照 cate2 的字典序排列，最后按照标题的字典序排列的

应该是这样吧

**Reply by carpedx** [reply]
想按创建时间排，不知道有没有这字段？ 看博主怎么回复把~

**Reply by mzlogin** [reply]
目前的逻辑确实如 @lanruixiang 所说。

如果想按创建时间排，Jekyll 的 Collections 目前是没有创建时间这个字段的，有一种能实现的办法是你给每个 wiki 文件的头信息里添加一个表示创建时间的字段，然后调整下 pages/wiki.md 里的逻辑。

**Comment by sosbai** [comment]
你好，当我在config_yml里，把baseurl:改为baseurl:/blog/时，在本地部署一直报错：(E:/myblog/mzlogin.github.io-master/_config.yml): mapping values are not allowed in this context at line 5 column 4  请问是什么原因。期待您的回复

**Reply by sosbai** [reply]
已解决，/blog前面要加个空格

**Comment by sosbai** [comment]
大佬，本地部署时，只要baseurl这里有值，在本地就跑不起来

**Reply by mzlogin** [reply]
具体是什么现象？这个模板目前对 baseurl 有值的情况确实缺少兼容。

我这边的现象是，能跑起来，但是引用的静态资源文件路径都是错的。

**Comment by ahan-io** [comment]
大佬，请教一下，本地预览的时候，执行：bundle exec jekyll serve，

报错：

   GitHub Metadata: No GitHub API authentication could be found. Some fields may be missing or have incorrect data.
  Liquid Exception: GET https://api.github.com/repos/ahan-ai/openbilibili/releases?per_page=100: 451 - Repository access blocked in /_layouts/page.html
jekyll 3.9.4 | Error:  GET https://api.github.com/repos/ahan-ai/openbilibili/releases?per_page=100: 451 - Repository access blocked
/Users/bytedance/.gem/ruby/3.1.3/gems/octokit-4.25.1/lib/octokit/response/raise_error.rb:14:in `on_complete': GET https://api.github.com/repos/ahan-ai/openbilibili/releases?per_page=100: 451 - Repository access blocked (Octokit::UnavailableForLegalReasons)

有什么思路没有？

**Reply by mzlogin** [reply]
你是在博客的什么地方引用到了 ahan-ai/openbilibili 这个仓库吗？这个仓库无法正常访问到

**Comment by zimoli-lamda** [comment]
你好大佬，我有如下几个问题自己弄了一天了，还是没有解决！

问题1：分类和关于这两个部分，页面上方的颜色怎么改成和主页一样的？

问题2：关于里面，我不想要双栏，也就是想把关于部分的Search、Categories Cloud、My Popular Repositories删除，只留下内容

问题3：开源那里，可以使用另外一个账号的git仓库吗？怎么改呢

**Reply by mzlogin** [reply]
1. 关于和分类页面上方的颜色，是根据 url 的字符动态计算的，如果要改成固定的颜色：
    - 关于页面：在第 2 点里面讲
    - 分类页面：修改 _layouts/categories.html 第 9 行 section 元素的样式，去掉 class `geopattern`，添加背景颜色样式；
2. 在 _layouts 目录里新建一个页面模板文件，不采用分栏布局，将 pages/about.md 文件头部的 layout 修改为新的页面模板的名字；自行定义新的页面模板的头部元素的背景颜色样式；
3. 是指 open-source 页面吗？我不知道是否可以，应该不能很简单地改个配置就实现，得自己想办法通过 js 调用 GitHub API 做了。

感觉你需要了解更多关于 GitHub Pages 和 Jekyll、以及 HTML 与 CSS 的知识，以下链接供参考：

- GitHub Pages：https://docs.github.com/zh/pages/quickstart
- Jekyll：https://jekyllrb.com/
- HTML 与 CSS：https://www.w3school.com.cn/h.asp

**Reply by zimoli-lamda** [reply]
好的好的，非常感谢您大佬！！！

**Comment by ghost** [comment]
你好大佬，我的网站有些错误：<zjx-kimi.github.io>
1. 博客无法显示
2. 无法回复
如果您方便的话可以 fork 帮我改一下吗？

**Reply by ghost** [reply]
现在ok了，但是还是无法回复+latex。

**Reply by mzlogin** [reply]
评论区配置参考：https://giscus.app/zh-CN ，对应修改 _config.yml 文件里的 giscus 相关配置项。

latex 我感觉像是你写的语法它不兼容？

我看你写的都是一个 `$` 包裹的，我这边以前测试是两个 `$$` 包裹，如：

```
When $$(a \ne 0)$$, there are two solutions to $$(ax^2 + bx + c = 0)$$ and they are

$$x = {-b \pm \sqrt{b^2-4ac} \over 2a}.$$
```

**Reply by ghost** [reply]
谢谢

**Comment by zackzheng1121** [comment]
大佬好，我覺得可能是我哪裡寫錯了，我在寫網誌的時候寫了對應的標籤和副標題的內容但是它不會正常顯示，會在文章前面顯示我所打的
像是這一篇：https://zackzheng1121.github.io//2025/03/30/%E7%AC%AC%E4%BA%8C%E9%A1%8C/
他不會有任何副標題和標籤，反而會顯示layout: post title: 第二題 categories: csharp description: 費氏數列計算器 keywords: csharp，請問大佬這要怎麼處理？十分感謝！

**Reply by mzlogin** [reply]
我看了下，是换行符的问题导致的。参考 <https://github.com/mzlogin/mzlogin.github.io/issues/47>

手动处理历史文章可以使用 `dos2unix` 命令。一劳永逸的做法是将 git 配置成提交时自动处理换行符，可以试试 `git config --global core.autocrlf input` 这个命令。

**Comment by zackzheng1121** [comment]
感謝大佬

Zhuang Ma ***@***.***> 於 2025年3月31日 週一 09:47 寫道：
